client.go

```diff
@@ -134,20 +134,23 @@ func (c *Client) Loop() bool {
 		conn.NotifyClose(chanErr)
 		conn.NotifyBlocked(chanBlocking)
 
-		select {
-		case err1 := <-chanErr:
-			c.reportErr(err1)
-
-			if conn1 := c.conn.Load().(*amqp.Connection); conn1 != nil {
-				c.conn.Store((*amqp.Connection)(nil))
-				conn1.Close()
-			}
-		case blocking := <-chanBlocking:
+		// loop for blocking/deblocking
+		for {
 			select {
-			case c.blocking <- blocking:
-			default:
+			case err1 := <-chanErr:
+				c.reportErr(err1)
+
+				if conn1 := c.conn.Load().(*amqp.Connection); conn1 != nil {
+					c.conn.Store((*amqp.Connection)(nil))
+					conn1.Close()
+				}
+				// return from routine to launch reconnect process
+				return
+			case blocking := <-chanBlocking:
+				c.blocking <- blocking
```

Comment:
this will block forever in case if user is not reading `c.blocking`, previously it had fallback with default case

		case blocking := <-chanBlocking:
			select {
			case c.blocking <- blocking:
			default:
			}
		}